geomstats/geometry/product_riemannian_metric.py

```diff
@@ -261,11 +265,14 @@ def log(self, point, base_point=None, point_type=None):
             return logs
 
         if point_type == 'matrix':
-            point = gs.to_ndarray(point, to_ndim=3)
-            base_point = gs.to_ndarray(base_point, to_ndim=3)
-            return gs.stack(
+            point = gs.to_ndarray(point, to_ndim=2, axis=0)
```

Comment:
shouldn't we use `riemannian_metric` if there is only one metric in the product to rule out this case?

Reply:
Yes! Although I'll leave that for a next PR if this is OK, this the goal of this one is not to clean this file but not use vectorization.